api/v1beta1/azureclusteridentity_types.go

```diff
@@ -43,17 +43,20 @@ type AllowedNamespaces struct {
 
 // AzureClusterIdentitySpec defines the parameters that are used to create an AzureIdentity.
 type AzureClusterIdentitySpec struct {
-	// UserAssignedMSI or Service Principal
+	// Type is the type of Azure Identity used.
```

Comment:
```suggestion
	// IdentityType is the type of Azure Identity used.
```

Although since this field is just `type` in its JSON representation, making this a proper Go comment here will create a naming discrepancy in the generated CRD description. I still think that would be more correct, but I am not invested in this change. 😄

Reply:
`Type` is the name of the field too... `IdentityType` is the type of the field (this is confusing!) :D 

So I think `Type` is more correct

Reply:
You're right--so hard to read when things "stutter" like that.